azext_iot/tests/helpers.py

```diff
@@ -189,26 +189,30 @@ def assign_role_assignment(
     scope: str,
     assignee: str,
     max_tries=10,
-    wait=10
-):
+    wait=10,
+) -> Optional[EmbeddedCLI]:
     """
     Assign rbac permissions to resource.
     """
     output = None
     tries = 0
     while tries < max_tries:
         role_assignments = get_role_assignments(scope=scope, role=role)
-        role_assignment_principal_ids = [assignment["principalId"] for assignment in role_assignments]
-        if assignee in role_assignment_principal_ids:
-            break
+        logger.info(f"Role assignments for the role of '{role}' against scope '{scope}': {role_assignments}")
+        role_assignment_principal_ids = [assignment.get("principalId") for assignment in role_assignments]
+        role_assignment_principal_names = [assignment.get("principalName") for assignment in role_assignments]
+        if assignee in role_assignment_principal_ids or assignee in role_assignment_principal_names:
+            return output
         # else assign role to scope and check again
         output = cli.invoke(
```

Comment:
Hi Paymaun, I found a underlying issue being discovered since now you changed the return type of get assignment to `List[dict]`. I found directly save `EmbeddedCLI` in variable will make the variable update everytime when cli.invoke call again, therefore when the loop run second time to check if the role assigned, the `output` will update to list assignment. Therefore I made some [suggestion change](https://github.com/Elsie4ever/azure-iot-cli-extension/commit/e483c3fe80ff31fa5b81e49e3a55e29bcc525cfa) here and it should be good. Sorry for the inconvenience!

Reply:
We discussed and resolved offline.